Resources

JessSumner opened the pull request:
@MattMSumner @halogenandtoast.
If you fancy taking a look at these updates I would be very grateful. Thank you both.

JessSumner committed:
Resources controller added.

Why:

This keeps my controllers small and with solo purposes. Allows me to use the resources helper for routes. Separates my views and css in a much clearer way.

How:

app/controllers/resources_controller.rb, app/views/resources/index.html.erb and app/assets/stylesheets/resources.css.scss have been created and utilized.

JessSumner committed:
Scroll links added to resources page.

Animation not added yet but resources.coffee file has been created.

JessSumner committed:
Reduced size of resources.html.erb

How:

Each section was removed into it's own partial.

JessSumner committed:
Formatted boxes for each resource item.

How

All boxes are the same size (except one). They are now set in rows of three so each box is wider. Box-shadow has been used to give a nice effect.

JessSumner committed:
Added information to each resource

Why:
Checked that all resources are correct and up to date. Information added included addresses which will be useful for the user. Added space for a map view so that users can get a visual cue for each resource. Started to move each resource information into a hash. This will reduce the size of the view file.

How:
Information for each resource is now going to be stored in an array of hashes in the controller.

JessSumner committed:
Added maps to resource cards

Why:

This gives a quick visual reference for the user.

How:

Used the google maps embed link. Also renamed a few of the CSS classes to be more descriptive and specific. Block became resource-cards. Text became vet-details-left and right.

JessSumner committed:
Resource cards cleaned up.

How:

Maps were positioned to be fixed to the bottom of the resource card. Height of maps was also fixed for a uniform look.
New css class resource-cards-small was created for rat sitting and rat guide links. This allows for the same style to be carried through with adaptions for the reduced content.

JessSumner committed:
Resources information extracted into YAML files

Why:
The controller was too large holding all the details.

How:
Resources folder made in Config which holds a YAML file for each vet section and the pharmacy details. The do loops in the views now call the string rather than the symbol. i.e vet["name"].

JessSumner committed:
Syntax corrections suggested by Hound

How:
Used double quotes when calling the YAML file. Added spaces between function arguments in css. Corrected the margin properties to be more concise. Removed the px from several 0px.

MattMSumner commented on app/views/resources/_rat_guides.html.erb:
```diff
@@ -0,0 +1,44 @@
+<h2 id="scroll-link-6" class="resource-title">Rat guides</h2>
+<ul class="resources">
+  <li class="resource-cards-small">
+    <h3> Rat Guide </h3>
```
How would you feel about pulling these into a yml file as well?

MattMSumner commented on app/views/resources/_rat_sitting.html.erb:
```diff
@@ -0,0 +1,12 @@
+<ul class ="resources">
+  <li class="resource-cards-small">
+    <h3>Benjamin Benefiel</h3>
+    <p>benjamin.benefiel@gmail.com</p>
```
And maybe these?

MattMSumner commented:
two question about yml files otherwise this LGTM

JessSumner committed:
Rat_guides and Rat_sitting pulled out to YAML

Why:

All information for all resources is now consistent and easy to find.

How:

Two new YAML files created and two do loops have been created in the view to iterate over all the resources.
